src/main/java/org/apache/commons/cli/DefaultParser.java

```diff
@@ -580,7 +580,7 @@ private boolean isArgument(final String token) {
      * Tests if the specified token is a Java-like property (-Dkey=value).
      */
     private boolean isJavaProperty(final String token) {
-        final String opt = token.substring(0, 1);
+        final String opt = token.length() > 0? token.substring(0, 1): null;
```

Comment:
Use String#isEmpty and flip the ternary. A space is missing around the "?". Add a test.

Reply:
Ping?

Reply:
I'm starting to resume the work as I've been not feeling well.

Reply:
> Ping?

I'm starting to resume the work as I've been not feeling well.

Reply:
Ping?